examples/gadgets/grpc/custom_operator/main.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	igjson "github.com/inspektor-gadget/inspektor-gadget/pkg/datasource/formatters/json"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+	grpcruntime "github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/grpc"
+)
+
+func do() error {
+	// Define an operator that will be executed on the data received from the server.
+	// In this example we implement our own operator, but it's also possible to use existing operators
+	// like the cli operator.
+	const opPriority = 50000
+	myOperator := simple.New("myHandler", simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
+		for _, d := range gadgetCtx.GetDataSources() {
+			jsonFormatter, _ := igjson.New(d)
+
+			d.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				jsonOutput := jsonFormatter.Marshal(data)
+				fmt.Printf("%s\n", jsonOutput)
+				return nil
+			}, opPriority)
+		}
+		return nil
+	}))
+
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT)
+	defer stop()
```

Comment:
Why this is needed here and not in other examples?

Reply:
It's needed to close the connection in a clean way, otherwise the server will print something like:

`WARN[0040] error on connection: rpc error: code = Canceled desc = context canceled`

I added a comment.